lib/src/regular_expressions/regular_expressions_extensions.dart

```diff
@@ -241,6 +245,61 @@ class RegExpComposer {
     return matches;
   }
 
+  static NlpMatch? matchBegin(RegExp regExp, String text, [bool trim = false]) {
+    final match = getMatchesSimple(regExp, text).firstOrNull;
+    if (match == null) {
+      return null;
+    }
+
+    String strBefore = text.substring(0, match.index);
+    if (trim) {
+      strBefore = strBefore.trim();
+    }
+
+    if (strBefore.isNotEmpty) {
+      return null;
+    }
+
+    return match;
+  }
+
+  static NlpMatch? matchEnd(RegExp regExp, String text, [bool trim = false]) {
```

Comment:
I thought I already created a method like this somewhere. Are you sure we need a new one?